### PR TITLE
Deserialization timeout

### PR DESCRIPTION
**Pull request by marcocast**
Adding timeout when serializing or deserializing variables

### CODE REVIEW

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 15 issues
* MAJOR 9 major
* MINOR 6 minor

#### Top 10 issues

1. MAJOR [AllObjects2BytesConverterHandler.java#L44](https://github.com/ow2-proactive/scheduling/blob/a10e2b776633bc658b8bc47b62ca9f696f9177e4/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L44): Add a private constructor to hide the implicit public one. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1118)
1. MAJOR [AllObjects2BytesConverterHandler.java#L76](https://github.com/ow2-proactive/scheduling/blob/a10e2b776633bc658b8bc47b62ca9f696f9177e4/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L76): Either log or rethrow this exception. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. MAJOR [AllObjects2BytesConverterHandler.java#L78](https://github.com/ow2-proactive/scheduling/blob/a10e2b776633bc658b8bc47b62ca9f696f9177e4/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L78): Remove this call to "exit" or ensure it is really required. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1147)
1. MAJOR [AllObjects2BytesConverterHandler.java#L101](https://github.com/ow2-proactive/scheduling/blob/a10e2b776633bc658b8bc47b62ca9f696f9177e4/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L101): Either log or rethrow this exception. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. MAJOR [AllObjects2BytesConverterHandler.java#L104](https://github.com/ow2-proactive/scheduling/blob/a10e2b776633bc658b8bc47b62ca9f696f9177e4/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L104): Remove this call to "exit" or ensure it is really required. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1147)
1. MAJOR [AllObjects2BytesConverterHandler.java#L122](https://github.com/ow2-proactive/scheduling/blob/a10e2b776633bc658b8bc47b62ca9f696f9177e4/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L122): Add the "@Override" annotation above this method signature [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)
1. MAJOR [AllObjects2BytesConverterHandler.java#L139](https://github.com/ow2-proactive/scheduling/blob/a10e2b776633bc658b8bc47b62ca9f696f9177e4/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L139): Add the "@Override" annotation above this method signature [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)
1. MAJOR [AllObjects2BytesConverterHandler.java#L152](https://github.com/ow2-proactive/scheduling/blob/a10e2b776633bc658b8bc47b62ca9f696f9177e4/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L152): Add the "@Override" annotation above this method signature [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)
1. MAJOR [AllObjects2BytesConverterHandler.java#L160](https://github.com/ow2-proactive/scheduling/blob/a10e2b776633bc658b8bc47b62ca9f696f9177e4/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L160): Add the "@Override" annotation above this method signature [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)
1. MINOR [SerializationUtil.java#L57](https://github.com/ow2-proactive/scheduling/blob/a10e2b776633bc658b8bc47b62ca9f696f9177e4/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/task/util/SerializationUtil.java#L57): Immediately return this expression instead of assigning it to the temporary variable "deserialized". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1488)

**Comment by paraita** [comment]
the git history is messy, can you squash the duplicate entries ?

**Comment by marcocast** [comment]
@paraita yes i ll squash all

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 15 issues
* MAJOR 9 major
* MINOR 6 minor

#### Top 10 issues

1. MAJOR [AllObjects2BytesConverterHandler.java#L44](https://github.com/ow2-proactive/scheduling/blob/66ac13edecad6f277d699aef654d1ae7f13c9ff6/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L44): Add a private constructor to hide the implicit public one. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1118)
1. MAJOR [AllObjects2BytesConverterHandler.java#L76](https://github.com/ow2-proactive/scheduling/blob/66ac13edecad6f277d699aef654d1ae7f13c9ff6/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L76): Either log or rethrow this exception. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. MAJOR [AllObjects2BytesConverterHandler.java#L78](https://github.com/ow2-proactive/scheduling/blob/66ac13edecad6f277d699aef654d1ae7f13c9ff6/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L78): Remove this call to "exit" or ensure it is really required. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1147)
1. MAJOR [AllObjects2BytesConverterHandler.java#L101](https://github.com/ow2-proactive/scheduling/blob/66ac13edecad6f277d699aef654d1ae7f13c9ff6/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L101): Either log or rethrow this exception. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. MAJOR [AllObjects2BytesConverterHandler.java#L104](https://github.com/ow2-proactive/scheduling/blob/66ac13edecad6f277d699aef654d1ae7f13c9ff6/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L104): Remove this call to "exit" or ensure it is really required. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1147)
1. MAJOR [AllObjects2BytesConverterHandler.java#L122](https://github.com/ow2-proactive/scheduling/blob/66ac13edecad6f277d699aef654d1ae7f13c9ff6/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L122): Add the "@Override" annotation above this method signature [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)
1. MAJOR [AllObjects2BytesConverterHandler.java#L139](https://github.com/ow2-proactive/scheduling/blob/66ac13edecad6f277d699aef654d1ae7f13c9ff6/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L139): Add the "@Override" annotation above this method signature [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)
1. MAJOR [AllObjects2BytesConverterHandler.java#L152](https://github.com/ow2-proactive/scheduling/blob/66ac13edecad6f277d699aef654d1ae7f13c9ff6/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L152): Add the "@Override" annotation above this method signature [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)
1. MAJOR [AllObjects2BytesConverterHandler.java#L160](https://github.com/ow2-proactive/scheduling/blob/66ac13edecad6f277d699aef654d1ae7f13c9ff6/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L160): Add the "@Override" annotation above this method signature [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)
1. MINOR [SerializationUtil.java#L57](https://github.com/ow2-proactive/scheduling/blob/66ac13edecad6f277d699aef654d1ae7f13c9ff6/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/task/util/SerializationUtil.java#L57): Immediately return this expression instead of assigning it to the temporary variable "deserialized". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1488)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 16 issues
* MAJOR 8 major
* MINOR 8 minor

#### Top 10 issues

1. MAJOR [AllObjects2BytesConverterHandler.java#L85](https://github.com/ow2-proactive/scheduling/blob/c9b853b77f9a4d8b7734dbc26a226b2d52d77bde/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L85): Either log or rethrow this exception. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. MAJOR [AllObjects2BytesConverterHandler.java#L87](https://github.com/ow2-proactive/scheduling/blob/c9b853b77f9a4d8b7734dbc26a226b2d52d77bde/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L87): Remove this call to "exit" or ensure it is really required. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1147)
1. MAJOR [AllObjects2BytesConverterHandler.java#L110](https://github.com/ow2-proactive/scheduling/blob/c9b853b77f9a4d8b7734dbc26a226b2d52d77bde/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L110): Either log or rethrow this exception. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. MAJOR [AllObjects2BytesConverterHandler.java#L112](https://github.com/ow2-proactive/scheduling/blob/c9b853b77f9a4d8b7734dbc26a226b2d52d77bde/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L112): Remove this call to "exit" or ensure it is really required. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1147)
1. MAJOR [AllObjects2BytesConverterHandler.java#L130](https://github.com/ow2-proactive/scheduling/blob/c9b853b77f9a4d8b7734dbc26a226b2d52d77bde/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L130): Add the "@Override" annotation above this method signature [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)
1. MAJOR [AllObjects2BytesConverterHandler.java#L147](https://github.com/ow2-proactive/scheduling/blob/c9b853b77f9a4d8b7734dbc26a226b2d52d77bde/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L147): Add the "@Override" annotation above this method signature [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)
1. MAJOR [AllObjects2BytesConverterHandler.java#L160](https://github.com/ow2-proactive/scheduling/blob/c9b853b77f9a4d8b7734dbc26a226b2d52d77bde/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L160): Add the "@Override" annotation above this method signature [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)
1. MAJOR [AllObjects2BytesConverterHandler.java#L168](https://github.com/ow2-proactive/scheduling/blob/c9b853b77f9a4d8b7734dbc26a226b2d52d77bde/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L168): Add the "@Override" annotation above this method signature [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1161)
1. MINOR [AllObjects2BytesConverterHandler.java#L47](https://github.com/ow2-proactive/scheduling/blob/c9b853b77f9a4d8b7734dbc26a226b2d52d77bde/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L47): Reorder the modifiers to comply with the Java Language Specification. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AModifiersOrderCheck)
1. MINOR [AllObjects2BytesConverterHandler.java#L49](https://github.com/ow2-proactive/scheduling/blob/c9b853b77f9a4d8b7734dbc26a226b2d52d77bde/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java#L49): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00115)

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                    byte[] serialized = serialiseValue(key, value);
+                    serializedMap.put(key, serialized);
+                }
+                return serializedMap;
+            }
+
+        };
+    }
+
+    private static Callable<Map<String, Serializable>> createDeserializeCollable(final Map<String, byte[]> target,
```
Comment (fviale) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java]: typo in the method name createDeserializedCallable

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                for (Entry<String, byte[]> e : target.entrySet()) {
+                    deserializedMap.put(e.getKey(),
+                                        (Serializable) Object2ByteConverter.convertByte2Object(e.getValue(), cl));
+                }
+                return deserializedMap;
+            }
+        };
+    }
+
+    private static Callable<byte[]> createSerializeSingleValueCollable(final String key, final Serializable value) {
```
Comment (fviale) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java]: same remark as above

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java (hunk 1 above)
Comment (fviale) [reply scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java]: or createCallableDeserializer

**Pull request merged**